### ISSUE TITLE
Improper optimizer load states with LR Find

### ISSUE THREAD

**Issue by muellerzr** [fastai/fastai#3185]
Despite `learn.lr_find()` zeroing the gradients of the optimizer (and subsequently loading in the optimizer), the state dict still manages to contain new gradient weights when loaded in. To recreate simply fit for an epoch and try running two subsequent LR find's:

```python
from fastai.vision.all import *
set_seed(99, True)
path = untar_data(URLs.PETS)/'images'
dls = ImageDataLoaders.from_name_func(
    path, get_image_files(path), valid_pct=0.2,
    label_func=lambda x: x[0].isupper(), item_tfms=Resize(224))

learn = cnn_learner(dls, resnet18)
learn.fit(1)
learn.lr_find()
learn.lr_find()
```

![image](https://user-images.githubusercontent.com/7831895/105560040-5e512680-5cd8-11eb-8626-8a55ecc8c47c.png)

You can also tell as `learn.opt.state_dict()` contains the grad's:
```python
learn.opt.state_dict()['state'][1]['grad_avg']
---
tensor([-6.0008e-03,  3.6600e-03,  0.0000e+00,  2.7814e-03,  0.0000e+00,
        -1.0096e-02,  2.2897e-03,  0.0000e+00, -8.1023e-03,  0.0000e+00,
         6.9529e-03,  2.5071e-03,  3.8387e-04,  0.0000e+00, -2.5570e-02,
         1.1745e-02,  4.7019e-03, -2.7299e-03, -1.0885e-03,  4.9580e-03,
         1.6914e-02, -2.2379e-04,  7.7674e-03, -5.6958e-03,  2.6118e-03,
         1.4418e-03,  3.4755e-04,  4.8158e-03,  6.2891e-03,  3.3802e-03,
         3.6270e-03,  1.4438e-04, -1.0190e-05,  5.7115e-03, -3.6538e-03,
         3.1913e-04,  0.0000e+00,  5.6190e-03,  0.0000e+00, -6.5286e-03,
         5.8704e-03,  1.2135e-03,  4.1734e-03, -1.3878e-03, -1.5321e-02,
        -9.7374e-03,  2.5666e-03,  9.3485e-03,  0.0000e+00, -6.9394e-03,
        -3.4711e-03,  1.9826e-02, -1.2847e-03,  8.3173e-04, -5.1534e-03,
        -2.1612e-02,  1.1994e-02,  9.0230e-03, -1.0857e-02,  7.3104e-03,
        -1.2067e-02,  1.4655e-03,  1.2315e-03, -6.2386e-03], device='cuda:0')
```
Whereas for example if this were a fresh model, this should return a key error (as there are no gradients yet)

**Comment by jph00** [comment]
Seems to be behaving correctly
